fabric-key-binding-api-v1/src/main/java/net/fabricmc/fabric/api/client/keybinding/v1/FabricKeyBinding.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.keybinding.v1;
+
+import java.util.Objects;
+import java.util.function.BooleanSupplier;
+
+import com.google.common.base.Preconditions;
+
+import net.minecraft.client.options.KeyBinding;
+import net.minecraft.client.util.InputUtil;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.impl.client.keybinding.StickyFabricKeyBinding;
+import net.fabricmc.fabric.mixin.client.keybinding.KeyCodeAccessor;
+
+/**
+ * Expanded version of {@link KeyBinding} for use by Fabric mods.
+ *
+ * <p>*ALL* instantiated FabricKeyBindings should be registered in
+ * {@link KeyBindingRegistry#register(FabricKeyBinding)}!
+ * </p>
+ *
+ * <pre><code>
+ * FabricKeyBinding.Builder builder = FabricKeyBinding.builder();
+ * FabricKeyBinding left = builder
+ * 			.id(new Identifier("example", "left"))
+ * 			.code(Keys.Left)
+ * 			.build();
+ * FabricKeyBinding right = builder
+ * 			.id(new Identifier("example", "right"))
+ * 			.code(Keys.Right)
+ * 			.build();
```

Comment:
Is there a specific reason to reuse the builder instance? It may be nicer to supply the required parameter (id) on creation, it is only one after all and recreating the builder doesn't look too bad to me.

Reply:
you shouldn't reuse the builder instance, will update this example